### ISSUE TITLE
ecs-cli compose service up giving error for --enable-service-discovery

### ISSUE THREAD

**Issue by shahbaz07** [aws/amazon-ecs-cli#1069]
Hi,

I am running below command to create service in Fargate and enable service discovery for it. It works fine and my service is discoverable in my VPC.

**ecs-cli compose --project-name backend service up --private-dns-namespace tutorial --vpc vpc-04deee8176dce7d7d --enable-service-discovery**

I have to run this command from my CD pipeline so if there is any change in service it will automatically deploy new version of service. But when I run this command again. It gives below error.

**Service Discovery can not be enabled on an existing ECS Service**

Please let me know how I can fix this issue. I saw in console there is option to choose existing service discovery for service. But I couldn't fine any option in ecs-cli compose service up documentation for this.

If you need any other information please let me know.

ecs-cli version 1.19.1 (a20cf74)
aws cli version: aws-cli/2.0.20 Python/3.7.3 Linux/4.4.0-1109-aws botocore/2.0.0dev24

Regards
Shahbaz Sajjad

**Comment by bvtujo** [comment]
Hi, thanks so much for bringing this up. I think this has to do with how `ecs-cli compose service up` calls the CreateService endpoint if the service doesn't exist and the UpdateService endpoint if the service does exist. We'll work on getting a fix out soon pending further investigation.

**Comment by shahbaz07** [comment]
Hi @bvtujo thanks for your reply. Please update me once this issue fixed. If you need any further information from me please let me know.

**Comment by bvtujo** [comment]
Hi @shahbaz07, we've looked into this and here's what we've found. 

`ecs-cli compose service up` calls either `CreateService` or `UpdateService` in the ECS API, depending on whether the service already exists. The reason you're getting an error when specifying `compose service up --enable-service-discovery` is that this field is only available on *creation* of a new service, not *updating* an existing one. Our default behavior here is to raise an error if a service already exists and the customer attempts to update fields which can only be specified on creation. 

I'm currently working on a possible fix, but was wondering if we could get some more details on your use case to see if we can avoid introducing a backward breaking change. Specifically, how does your pipeline operate? What guarantees are made about the state of the service at a given pipeline stage? 

Is it possible to separate out your calls which require `--enable-service-discovery` or `--scheduling-strategy` into a logic block, based on e.g. the output of `ecs-cli compose service ps`? This should list all the containers started by a given service.

Alternatively, is it possible for you to separately call one of the following options?
```bash
ecs-cli compose --project-name backend service up --private-dns-namespace tutorial --vpc vpc-04deee8176dce7d7d --enable-service-discovery
ecs-cli compose --project-name backend service start
```
or 
```bash
ecs-cli compose --project-name backend service scale 1
```
For more information on the compose subcommands, you can check the out our [documentation](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cmd-ecs-cli-compose-service.html#cmd-ecs-cli-compose-service-subcommands)

**Comment by shahbaz07** [comment]
Hi @bvtujo,

Thanks for your detailed reply. We are currently using gitlab ci for our pipeline. We excute above commands in script when something merge to specific branch. We can write a script as per your suggestion which can check whether that service is running before or not. But we run above command with --create-log-groups but this flag only give warning. That log group already exist. So I think if you guys can have same check if we run this command with --enable-service-discovery it check if service discovery is already there for this service do not create and use existing one and update the service. If it's not possible or very difficult to implement we will write a script as per your suggestion we will first execute service ps command and check if service exist we will remove this flag from service up command.

Please let me know if you need any other information. Also please let me know where we can ask some questions related to AWS Fargate and ecs-cli.

**Comment by bvtujo** [comment]
@shahbaz07 We've completed a fix for this which will be released in 1.20.0 shortly. This downgrades the error received when specifying `--enable-service-discovery` on a call to UpdateService to a warning, and should unblock your CI pipeline. Please let us know if you have any further questions.

**Comment by shahbaz07** [comment]
Hi @bvtujo, thanks for the fix I will check this if I will face any issue I will let you know.

Please let me know where we can ask some questions related to AWS Fargate and ecs-cli.
